src/w_wad.c

```diff
@@ -91,90 +92,112 @@ static int *handles = NULL;
 
 static void W_AddFile(const char *name) // killough 1/31/98: static, const
 {
-  wadinfo_t   header;
-  lumpinfo_t* lump_p;
-  unsigned    i;
-  int         handle;
-  int         length;
-  int         startlump;
-  filelump_t  *fileinfo, *fileinfo2free=NULL; //killough
-  filelump_t  singleinfo;
-  boolean     is_single = false;
-  char        *filename = strcpy(malloc(strlen(name)+5), name);
-
-  NormalizeSlashes(AddDefaultExtension(filename, ".wad"));  // killough 11/98
-
-  // open the file and add to directory
-
-  if ((handle = M_open(filename,O_RDONLY | O_BINARY)) == -1)
+    wadinfo_t header;
+    filelump_t *fileinfo;
+    int handle;
+    int length;
+    int startlump;
+
+    // open the file and add to directory
+
+    handle = M_open(name, O_RDONLY | O_BINARY);
+
+    if (handle == -1)
     {
-      if (strlen(name) > 4 && !strcasecmp(name+strlen(name)-4 , ".lmp" ))
-	{
-	  free(filename);
-	  return;
-	}
-      // killough 11/98: allow .lmp extension if none existed before
-      NormalizeSlashes(AddDefaultExtension(strcpy(filename, name), ".lmp"));
-      if ((handle = M_open(filename,O_RDONLY | O_BINARY)) == -1)
-	I_Error("Error: couldn't open %s\n",name);  // killough
+        if (M_StringCaseEndsWith(name, ".lmp"))
```

Comment:
Why do we accept non-existing .lmp files? Is this for `-playdemo`? It applies to `-file` as well.

Reply:
```
$ woof -file nothere
Error: couldn't open nothere

$ woof -file nothere.wad
Error: couldn't open nothere.wad

$ woof -file nothere.lmp
[...]
```

Reply:
> Why do we accept non-existing .lmp files? Is this for `-playdemo`? It applies to `-file` as well.

Yes, it's for -playdemo of built-in WAD demos.

Reply:
I still don't get it. Why do we check for the `.lmp` extension and not for no extension at all? I mean `-playdemo demo1` is supposed to work, not `-playdemo demo1.lmp` (well, if the file does not actually exist, that is).

Reply:
> Why do we check for the .lmp extension and not for no extension at all?

Because the .lmp extension has already been added somewhere in d_main or g_game. We should refactor this, but it's probably out of scope for this PR. At the moment I just copy the previous code for this case.

Reply:
Okay.

Reply:
> Because the .lmp extension has already been added somewhere in d_main

Here: https://github.com/fabiangreffrath/woof/blob/1ec38042a3dbd82f9467158af0d258de37eb6858/src/d_main.c#L2346